app/views/user/download.scala

```diff
@@ -15,9 +15,98 @@ object download {
       main(cls := "box page-small search")(
         h1(title),
         form(cls := "form3 box__pad")(
-          "The form here"
+          table(
+            tr(
+              th(label(cls := "form-label")(trans.username())),
+              td(st.input(tpe := "text", cls := "form-control", name := "username", value := s"${user.username}"))),
+
+            trParam("since", "Since", "date"),
+            trParam("until", "Until", "date"),
+            trParam("max", "Max", "input"),
+            trParam("opponent", trans.opponent.txt(), "input"),
+            trParam("rated", trans.rated.txt(), "checkbox"),
+            tr(thHeader("ratingCategoryO", "Rating Category"),
+              td(spanRatingPref("p-ultrabullet", "Ultrabullet"),
+                spanRatingPref("p-bullet", "Bullet"),
+                spanRatingPref("p-blitz", "Blitz"),
+                spanRatingPref("p-rapid", "Rapid"),
+                spanRatingPref("p-classical", "Classical"),
+                spanRatingPref("p-correspondence", "Correspondence"),
+                spanRatingPref("p-chess960", "Chess960"),
+                spanRatingPref("p-crazyhouse", "Crazyhouse"),
+                spanRatingPref("p-antichess", "Antichess"),
+                spanRatingPref("p-atomic", "Atomic"),
+                spanRatingPref("p-horde", "Horde"),
+                spanRatingPref("p-koth", "King of the Hill"),
+                spanRatingPref("p-racing-kings", "Racing Kings"),
+                spanRatingPref("p-three-check", "Three Check")
```

Comment:
Are the following game types not extracted as string constants somewhere in the codebase:
- Bullet
- Blitz
- Rapid
- Classical
- Correspondence
- Chess960
- Crazyhouse
- Antichess
- Atomic
- Horde
- King of the Hill
- Racing Kings
- Three Check

Reply:
They are: `lila.rating.PerfType.nonPuzzle` gives a list of the corresponding PerfType objects. Their `trans` attribute will give those strings in the user's language.